### DISCUSSION TITLE
MICRO AE.exp does not specify the llc_prefetcher_types

### DISCUSSION THREAD

**Discussion by szuhzw** [CMU-SAFARI/Hermes#7] [Q&A] [answered]
Hello. MICRO AE.exp does not specify the llc_prefetcher_types. Why does multi.llc_pref not output the message "unsupported prefetcher type "?

**Comment by szuhzw** [comment]
Can anyone resolve my doubts?

**Comment by rahulbera** [comment, marked as answer]
HI @szuhzw, the `llc_prefetcher_types` is a knob of type vector of strings (`src/knobs.cc:22`). If you have have not specify `llc_prefetcher_types` knob, the vector will be of size 0, in which case the code inside the `for` loop at `prefetcher/multi.llc_pref:26` will not even execute. Hence, there won't be any log saying "unsupported prefetcher types". The log will be printed _if and only if_ you have passed an invalid prefetcher name with `llc_prefetcher_types` knob.

Closing the issue for now. If needed, we can reopen.

**Reply by szuhzw** [reply]
Thank you.
